RecoEgamma/EgammaTools/src/GainSwitchTools.cc

```diff
@@ -0,0 +1,136 @@
+#include "RecoEgamma/EgammaTools/interface/GainSwitchTools.h"
+
+
+#include "DataFormats/EcalRecHit/interface/EcalRecHit.h"
+#include "Geometry/CaloTopology/interface/CaloTopology.h"
+#include "RecoCaloTools/Navigation/interface/CaloNavigator.h"
+#include "DataFormats/DetId/interface/DetId.h"
+const std::vector<int> GainSwitchTools::gainSwitchFlags_={EcalRecHit::kHasSwitchToGain6,EcalRecHit::kHasSwitchToGain1};
+
+ //this should really live in EcalClusterTools
+int GainSwitchTools::nrCrysWithFlagsIn5x5(const DetId& id,const std::vector<int>& flags,const EcalRecHitCollection* recHits,const CaloTopology *topology)
+{
+  int nrFound=0;
+  CaloNavigator<DetId> cursor = CaloNavigator<DetId>( id, topology->getSubdetectorTopology( id ) );
+  
+  for ( int eastNr = -2; eastNr <= 2; ++eastNr ) { //east is eta in barrel
+    for ( int northNr = -2; northNr <= 2; ++northNr ) { //north is phi in barrel
+      cursor.home();
+      cursor.offsetBy( eastNr, northNr);
+      DetId id = *cursor;
+      auto recHitIt = recHits->find(id);
```

Comment:
please review the "LogicError" issue related to this
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17296/17478/llvm-analysis/report-105848.html#EndPath
it could be false-positive, but better cross-check